### DISCUSSION TITLE
Ayuda con compilacion de libreria!!!

### DISCUSSION THREAD

**Discussion by luciano-repetti** [INGCOM-UNRN-P1/cursada-2023#231] [Preguntas y respuestas] [answered]
Buenas! Esta pregunta va mas que nada para el profe Martin ya que es sobre la implementación de librerias externas en C. Para la implementación de SQLite.

Lo que hice hasta ahora fue descargar [sqlite-amalgamation-3440000](https://www.sqlite.org/2023/sqlite-amalgamation-3440000.zip) y [sqlite-tools-win-x64-3440000](https://www.sqlite.org/2023/sqlite-tools-win-x64-3440000.zip)

Los descomprimi y tire todo en una carpeta llamada "SQLite" en mi disco local C. Puse la direccion en las variables del sistema en PATH para poder ejecutar sqlite3 desde cualquier lugar. Luego configure el "includePath" de c/c++ microsoft para poner la ruta de sqlite:

```
{
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**",
                "C:\\Program Files\\SQLite\\**"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],
            "compilerPath": "C:\\mingw64\\bin\\gcc.exe",
            "cStandard": "c17",
            "cppStandard": "gnu++17",
            "intelliSenseMode": "windows-gcc-x64"
        }
    ],
    "version": 4
}
```

Con esto ya se me fue el error de: #include errors detected. Please update your includePath. Squiggles are disabled for this translation unit

Pero al momento de compilar me dice lo siguiente:

```sqlite3.h: No such file or directorygcc```, no entiendo si tengo que descargar un archivo especifico o alguna configuracion más.

PD: Si funciona el comando sqlite3.

PD2: El que quiera consultar para implementarlo en su codigo o aportar a la discusión es más que bienvenido.

**Comment by martinvilu** [comment]
```
"C:\\Program Files\\SQLite\\**"
```
Esto es la invitación al desastre, además de problemas para la compilación, estás agregando archivos de usuario en una carpeta de sistema. 

Todo lo de usuario tiene que estar en tu usuario, muy en especial si no tiene instalador.

El `includepath` es para las ayudas del entorno, no para el compilador, por lo que no va a funcionar.

**Reply by luciano-repetti** [reply]
Okey, entonces cual es la mejor forma de hacerlo para implementar SQLite.

No sé si te acordás, pero habías dicho que era más fácil usar SQLite que implementar expresiones regulares. 

La idea es aprender más cosas por nuestra cuenta y no quedarnos en lo básico de la guía. Y respecto a esto no encontramos tanta información para poder implementarlo.

**Reply by martinvilu** [reply]
Tené en cuenta que tiene que compilar de una, ya que de otra forma, no pasa el primer filtro.

Lo que es de compilación, va con el compilador o en el proyecto. No lo vemos porque el compilador que estamos usando es más simple de instalar, pero todo lo extra es complicado.

**Reply by martinvilu** [reply]
Lo que hay que cambiar es la forma de llamar al compilador de forma que agregue esta librería. 

Pero también requiere que donde esté descompactado tenga permisos de ejecución y menos potencial a romper los programas de tu computadora.

**Reply by luciano-repetti** [reply]
Okey, entonces no lo implementamos nada la base de datos?

**Reply by martinvilu** [reply]
![austin-powers-dangerously](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/56625/3100be95-e7d3-4196-bf52-ebd3886b3e6b)

Hace dos preguntas, 

1. ¿es necesario? 
2. ¿Quiero aceptar el riesgo de que no compile al ser revisado?

**Reply by luciano-repetti** [reply, marked as answer]
Bueno haremos una versión 2 de última para poder aprender a utilizar librerías externas. Tal vez con un makefile que se ejecute antes del compilador o tener buildeado el proyecto y que directamente se ejecute.
